### PR TITLE
New hyperspy demo

### PR DESCRIPTION
**Pull request by bm424**
Created the `HSPYSVT` class inheriting from `SVT`. Main difference is that it should be able to take a HyperSpy signal and convert it to a numpy array of the appropriate dimensions/shape before passing it back to `SVT` before denoising. It _should_ then produce a new HyperSpy spectrum as output, but I can't test that. :neutral_face:

### CODE REVIEW

**Comment by tjof2** [comment]
Thanks Ben, I need to finish off the Python wrapper this week so will be able to test it then.

**Comment by tjof2** [comment]
I've now finished the basic Python wrapper (hence the conflicts - thanks for spotting the missing commas!). Will test this asap.

**Comment by tjof2** [comment]
@bm424 can you rebase this against the current master?

**Comment by tjof2** [comment]
@bm424  This doesn't seem to handle cases where the image sequence is loaded as a 3D TIFF stack? For example, TIFF stacks as saved from ImageJ.

``` python
movie = load("example_sequence.tif")
print movie.data.shape
(200, 128, 128)
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/hyperspy_demo.py", line 69, in <module>
    data = HSPYSVT.prepare_to_denoise(movie, spectral_axis=0)
  File "/hyperspy_demo.py", line 33, in prepare_to_denoise
    signal_data = signal_3d._data_aligned_with_axes
AttributeError: 'Spectrum' object has no attribute '_data_aligned_with_axes'
```

**Comment by tjof2** [comment]
@bm424  Sorry...current master has changed significantly again due to changes to CMake!

**Comment by bm424** [comment]
@tjof2 OK, to flesh out the commit message for ec0c302 a bit, I've gone and rewritten a fairly chunky part of the main wrapper I had already made, which means that literally  no user input _should_ be necessary (I've included some introspection to work out the correct transformations to go for in each case). I've also moved it into the same folder as the original `pguresvt.py` for consistency. I had to make a couple smallish changes to your code to avoid some conflicts:
- Somewhere in the install process, it clearly needs `pguresvt.py.in` to work - I couldn't work out what a `.py.in` file was, but python can't import them. Since you'll know better than I what is needed here, the easiest fix was a copy of the file as a normal `.py` file.
- It seems that the install method for PGURE-SVT doesn't account for the use of `virtualenv`s - the main problem being line 140 in the original [`pguresvt.py`](https://github.com/tjof2/pgure-svt/blob/master/pguresvt/pguresvt.py.in). I fixed this by [changing it to an absolute reference](https://github.com/bm424/pgure-svt/blob/NEW_hyperspy_demo/pguresvt/pguresvt.py) in the new `pguresvt.py`.
- Again for using `virtualenv`s, I've included a `requirements.txt` which _appears_ to be the minimum python dependency tree, (although I did have to manually install PyQT and SIP - that's a different issue)

To sum up: `pguresvt/hspysvt.py` now contains the main `HSPYSVT` class, and when run should successfully go through an example; `HyperSpy Demo.ipynb` contains a working example with some instructions; there appears to be an inconsistency with relative/absolute paths during the install of the module.

Hope all that's useful!

**Comment by tjof2** [comment]
@bm424  Thanks for spotting the virtualenvs.

As for the `.in` suffix - it's arbitrary, it's just a file that CMake copies to a `.py` file on building, but with some variables updated at build-time (e.g. version, timestamp, path to dll file etc.) - I'll fix this myself.

Thanks for all of this! Will test tomorrow :-)

**Comment by tjof2** [comment]
i.e. the variable `${PYTHONLIBRARYPATH}` in pguresvt.py.in _should_ change to the correct path on installation with CMake - i.e. it should point to the place where CMake puts the library. But I'll update this separately in CMake.

For example on my machine it points to /usr/lib.
